Many-body-guess

```diff
@@ -0,0 +1,31 @@
+from wrap_pyscf import *
+import numpy as np
+
+mol = [("H", -0.214000, 0.046200, -2.369500),
+       ("H", -1.056200, 0.297200, -3.031300),
+       ("H", -1.888200, 0.098300, -0.089800),
+       ("H", -1.661000, 1.114200, 0.265700),
+       ("H", 1.097200, 0.291800, 0.108100),
+       ("H", 0.911200, 0.254400, 1.191600)]
+
+bs = "sto-3g"
+
+h2A = density(mol[0:2], bs)
+h2B = density(mol[2:4], bs)
+h2C = density(mol[4:6], bs)
+h4AB = density(mol[0:4], bs)
+h4AC = density((mol[0:2]+mol[4:6]), bs)
+h4BC = density(mol[2:6], bs)
+#density_list = [h2A, h2B, h2C, h4AB, h4AC, h4BC]
+
+A = np.zeros((6,6))
+
+A[0:4, 0:4] = h4AB
+A[4:6, 2:6] = h4BC[2:4, 0:4]
+A[2:4, 4:6] = h4BC[0:2, 2:4]
+A[4:6, 0:2] = h4AC[2:4, 0:2]
+A[0:2, 4:6] = h4AC[0:2, 2:4]
```

Comment:
This doesn't look right to me. In particular you're overwriting AA blocks, and not adding to them, for example. Also you're not subtracting out the monomer densities.

Reply:
The final density is not what I put in the slack chat yesterday? My misunderstanding. I thought we had to subtract out an AA because you get two of those subblocks, from A-B density and A-C density. So i used full A-B, but only parts of B-C and A-C to avoid the overcounting of the monomer subblocks AA, BB, and CC. 

I thought this is what Jonathan was talking about.

Reply:
And so I thought that since there was no overcounting with this piecewise construction of the guess rho, no addition or subtraction.

Reply:
Where did my thinking go wrong here, if you don't mind saying?

Reply:
The way this is currently written loses some of the desired interactions. The interaction between A and B is mostly contained in the AB and BA blocks of the A-B density, but there are effects on the AA and BB parts as well. So the AA block of the guess density needs to AA_{A-B} + AA_{A-C} - AA_{A}, so that all desired interactions are captured in the guess.

You can use straight assignment for the off-diagonal blocks, but the arithmetic is required (in some fashion) for the diagonal blocks.